packages/core/src/components/button/button.scss

```diff
@@ -1,3 +1,209 @@
 :host {
-  display: block;
+  display: inline-block;
+  width: auto;
+  cursor: pointer;
+}
+
+.button {
+  box-sizing: border-box;
+  display: inline-flex;
+  align-items: stretch;
+  justify-content: center;
+  width: 100%;
+  border-style: solid;
+  border-width: var(--fl-input-border-width);
+  font-family: var(--fl-input-font-family);
+  font-weight: var(--fl-font-weight-semibold);
+  text-decoration: none;
+  user-select: none;
+  white-space: nowrap;
+  vertical-align: middle;
+  padding: 0;
+  transition: var(--fl-transition-fast) background-color, var(--fl-transition-fast) color,
+    var(--fl-transition-fast) border, var(--fl-transition-fast) box-shadow;
+  cursor: inherit;
+
+  &::-moz-focus-inner {
+    border: 0;
+  }
+
+  &:focus {
+    outline: none;
+  }
+
+  &.button--disabled {
+    opacity: 0.5;
+    cursor: not-allowed;
+
+    // When disabled, prevent mouse events from bubbling up
```

Comment:
why are we preventing mouse events from bubbling up?

Reply:
When the button is disabled, we don't want clicks that are on the button to trigger a form submission per say